protoc-gen-gogoroach/main.go

```diff
@@ -0,0 +1,41 @@
+// This is a customized version of https://github.com/gogo/protobuf/blob/master/protoc-gen-gogoslick/main.go
+
+package main
+
+import (
+	"github.com/gogo/protobuf/vanity"
+	"github.com/gogo/protobuf/vanity/command"
+)
+
+func main() {
+	req := command.Read()
+	files := req.GetProtoFile()
+
+	vanity.ForEachFile(files, vanity.TurnOnMarshalerAll)
+	vanity.ForEachFile(files, vanity.TurnOnSizerAll)
+	vanity.ForEachFile(files, vanity.TurnOnUnmarshalerAll)
+
+	// either one of these breaks compilation of the descriptor proto :(
```

Comment:
Typically I don't regenerate the descriptor.proto, but if you must it should be easy to write your own ForEachFile function that just skips the descriptor.proto